pkg/controller/vsphere/reconciler_test.go

```diff
@@ -783,8 +784,381 @@ func TestDelete(t *testing.T) {
+	credentialsSecretPassword := fmt.Sprintf("%s.password", server.URL.Host)
+
+	password, _ := server.URL.User.Password()
+	namespace := "test"
+	vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+
+	credentialsSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			credentialsSecretUsername: []byte(server.URL.User.Username()),
+			credentialsSecretPassword: []byte(password),
+		},
+	}
+
+	vmObj := object.NewVirtualMachine(session.Client.Client, vm.Reference())
+	task, err := vmObj.PowerOn(context.TODO())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []struct {
+		name                  string
+		expectedError         error
+		providerSpec          vsphereapi.VSphereMachineProviderSpec
+		labels                map[string]string
+		notConnectedToVCenter bool
+	}{
+		{
+			name: "Successfully create machine",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				Template: vm.Name,
+			},
+		},
+		{
+			name: "Fail on invalid missing machine label",
+			labels: map[string]string{
+				machinev1.MachineClusterIDLabel: "",
+			},
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				Template: vm.Name,
+			},
+			expectedError: errors.New("test: failed validating machine provider spec: test: missing \"machine.openshift.io/cluster-api-cluster\" label"),
+		},
+		{
+			name: "Fail on not connected to vCenter",
+			providerSpec: vsphereapi.VSphereMachineProviderSpec{
+				Template: vm.Name,
+			},
+			expectedError:         errors.New("test: not connected to a vCenter"),
+			notConnectedToVCenter: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			labels := map[string]string{
+				machinev1.MachineClusterIDLabel: "CLUSTERID",
+			}
+
+			if tc.labels != nil {
+				labels = tc.labels
+			}
+
+			if tc.notConnectedToVCenter {
+				session.Client.ServiceContent.About.ApiType = ""
+			}
+
+			machineScope := machineScope{
+				Context: context.TODO(),
+				machine: &machinev1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test",
+						Namespace: "test",
+						Labels:    labels,
+					},
+				},
+				providerSpec: &tc.providerSpec,
+				session:      session,
+				providerStatus: &vsphereapi.VSphereMachineProviderStatus{
+					TaskRef: task.Reference().Value,
+				},
+				client: fake.NewFakeClientWithScheme(scheme.Scheme, &credentialsSecret),
+			}
+
+			reconciler := newReconciler(&machineScope)
+
+			err = reconciler.create()
+
+			if tc.expectedError != nil {
+				if err == nil {
+					t.Error("reconciler was expected to return error")
+				}
+				if err.Error() != tc.expectedError.Error() {
+					t.Errorf("Expected: %v, got %v", tc.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("reconciler was not expected to return error: %v", err)
+				}
+			}
+		})
+	}
+}
+func TestUpdate(t *testing.T) {
```

Comment:
Nit: I'd add a new line above this one between these two tests

Reply:
👍